src/Checkbox.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import Icon from './Icon'
+import Flex from './Flex'
+import Box from './Box'
+import theme from './theme'
+
+class Checkbox extends React.Component {
+  constructor({ value, isChecked }) {
+    super()
+    this.state = {
+      value,
+      isChecked
+    }
+  }
+
+  handleClick() {
```

Comment:
Since the design system components are stateless, peophaps you can take a look at the ‘onClick’ implementation in ‘CloseBox’, for the similar click on the checkbox? It’s really up to the story or the client app to define the code for clicking on the checkbox, pagans?
https://github.com/pricelinelabs/design-system/blob/master/src/CloseButton.js#L13

Reply:
yeah, "checkAction"  is that function.  maybe I should rename it.  I have to have another function call it internally though. The checkbox you click on is really a facade, and I want to return the actual uncontrolled state, name, and value of the real hidden checkbox.